### PR TITLE
feat: Extract Optimize v3.8.0 docs to new /optimize section

### PR DESCRIPTION
**Pull request by pepopowitz**
Part of #1116 

~DO NOT MERGE until this PR has been reviewed, and all follow-up PRs (see below) have been reviewed & merged into this one.~

## What is the purpose of the change

This PR extracts version 3.8.0 of the Optimize docs from the 8.0 C8 docs, and into the new /optimize doc section. 

For the sake of review-ability, I'm planning to break down the work as follows: 

**This PR:**
- [x] Add version 3.8.0 to the `optimize` docs instance
- [x] Move 1 file (`what-is-optimize`) from docs v8.0 into optimize v3.8.0
- [x] Build out sidebars for optimize v3.8.0, with most links pointed at docs v8.0
- [x] Rewrite sidebar generation based on the fact that versioned sidebars can only use .json files, not .js

**Coming in another PR, but will be merged _into this one_ before merging this one:**
- [x] Customize the version selector in the navbar to use the versions associated with the current doc (in #1235)

**Coming in another PR, but will be merged _into this one_ before merging this one:**
- [x] Extract the remaining optimize 3.8.0 files from docs v8.0 (in #1236)

**Coming in another PR, but will be merged _into 1236_ before merging #1236 into this one:**
- [x] Add rewrite rules for all optimize 3.8.0 files (in #1238)

**Coming in another PR, but will be merged _into this one_ before merging this one:**
- [x] Put `docs/next/` and `optimize/` folders back in the sitemap (in #1239)

## PR Checklist

- [x] My changes apply to an already released version, and I have added them to the relevant `/versioned_docs` directory, or they are not for an already released version.
- [x] My changes apply to future versions, and I have added them to the main `/docs` directory, or they are not for future versions.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

**Comment by akeller** [comment]
I really like how transparent you've been about how you are breaking this work down across PRs. Thank you!

Taking a look at this now, expect a review from me shortly.

File: versioned_sidebars/version-8.0-sidebars.json [hunk 1]
```diff
@@ -291,7 +291,11 @@
         }
       ],
       "Optimize": [
-        "components/optimize/what-is-optimize",
+        {
+          "type": "link",
+          "label": "What is Optimize?",
+          "href": "/optimize/what-is-optimize"
```
Comment (akeller) [inline versioned_sidebars/version-8.0-sidebars.json:297]: I think this is why your build is failing - you'll need a redirect from `/components/optimize/what-is-optimize` to `/optimize/what-is-optimize`.

**Review by akeller — APPROVED** [review]
This is what I expected to see. Thank you for the well documented `generateOptimizeSidebars.js`! It was super interesting to read about js vs json too. I either didn't notice or hadn't fully realized it yet.

File: versioned_sidebars/version-8.0-sidebars.json (hunk 1 above)
Comment (pepopowitz) [reply versioned_sidebars/version-8.0-sidebars.json:297]: yeah...all these moved docs are going to fail on the link-check. I'll do all the redirects together in a future PR, and hopefully we'll see everything go green there!

**Review by akeller — APPROVED** [review]
I'm clicking around on my local copy and everything works as I expect it to!

**Pull request merged**